switchboard/src/routes/jobs.rs

```diff
@@ -0,0 +1,139 @@
+use crate::auth::db::DbAuth;
+use crate::auth::extract::AuthSource;
+use crate::auth::AuthorizationSource;
+use crate::perms::{ReadJobStatusError, StopJobError, SubmitJobError};
+use crate::routes::proxy::{proxy_err, proxy_val, Proxied};
+use crate::serve::AppState;
+use crate::{impl_from_auth_err, perms, sql};
+use axum::extract::{Path, State};
+use axum::Json;
+use futures_util::stream::FuturesOrdered;
+use futures_util::StreamExt;
+use std::str::FromStr;
+use treadmill_rs::api::switchboard::jobs::{
+    list::Response as LJResponse,
+    status::Response as JSResponse,
+    stop::Response as SJResponse,
+    submit::{Request as EJRequest, Response as EJResponse},
+};
+use uuid::Uuid;
+
+// -- submit
+
+impl_from_auth_err!(EJResponse, Database => Internal, Unauthorized => Unauthorized);
+#[tracing::instrument(skip(state, auth))]
+pub async fn submit(
+    State(state): State<AppState>,
+    AuthSource(auth): AuthSource<DbAuth>,
+    Json(request): Json<EJRequest>,
+) -> Proxied<EJResponse> {
+    let submit_job_priv = auth.authorize(perms::SubmitJob).await.map_err(proxy_err)?;
+    let user_id = submit_job_priv.subject().user_id();
+    let job_id = perms::submit_job(&state, submit_job_priv, request.job_request)
```

Comment:
The way this handles permissions is *really cool* and is exactly what I envisioned when we chatted a couple months ago @max-cura!

I do think that it's slightly weird to have some of the business logic be in the `perms` module (`perms::submit_job`), but we can move that later.